src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java

```
@@ -0,0 +1,386 @@
+            }
+
+        } catch (Exception e) {
+            log.error("Problem in publishing to message" + e);
+            throw new ConnectException(e);
+        }
+    }
+
+    /**
+     * Initialize queue
+     */
+    MQQueue setQueue(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
+        MQQueueManager manager = connectionBuilder.getManager();
+        MQQueue queue;
+        try {
+            queue = manager.accessQueue(config.getQueue(), CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating queue " + e);
+            return null;
+        }
+        return queue;
+    }
+
+    /**
+     * Create datagram or request MQMessage
+     */
+    MQMessage buildDatagramOrReplyMessage(MQConfiguration config, String queueMessage) {
+
+        MQMessage mqMessage = new MQMessage();
+        mqMessage.messageId = config.getMessageID().getBytes();
+        mqMessage.correlationId = config.getCorrelationID().getBytes();
+        log.info("Correlation ID " + config.getCorrelationID());
+        log.info("Message ID " + config.getMessageID());
+        if (config.getPriority() != -1000) {
+            mqMessage.priority = config.getPriority();
+        }
+
+        int messageType = config.getMessageType();
+        mqMessage.messageType = messageType;
+
+        if (config.isPersistent()) {
+            mqMessage.persistence = MQPER_PERSISTENT;
+        } else {
+            mqMessage.persistence = MQPER_NOT_PERSISTENT;
+        }
+        if (config.getgroupID() != null) {
+            mqMessage.groupId = config.getgroupID().getBytes();
+        }
+
+        GregorianCalendar cal = new GregorianCalendar();
+        cal.setTime(new Date(System.currentTimeMillis()));
+        mqMessage.putDateTime = cal;
+
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
+        }
+        return mqMessage;
```

I think for all these message types, it s better to throw the error to higher layer and from there it should be wrapped as ConnectionException. Returning message as empty and handle that as a generic one, one thing we loose information from lower layer why it is failed.